### ISSUE TITLE
Convert pre-trained .npz model to frozen graph in tensorflow?

### ISSUE THREAD

**Issue by WeiyiLi** [tensorpack/tensorpack#1062]
I would like to try a pre-trained model of shuffleNet, which is in https://github.com/tensorpack/tensorpack/tree/master/examples/ImageNetModels, in tensorflow. However, the format of this model are .npz. How can I convert to a frozen graph? Or how can I make tensorflow use this model for inference? Thanks!

**Comment by ppwwyyxx** [comment]
Closing due to inactivity.

**Comment by WeiyiLi** [comment]
Thanks. But how can I get the input names and output names? I am trying shufflenet V2.

**Comment by ppwwyyxx** [comment]
The easiest way is to `print(tensor)`. It will tell you the name of the tensor.

**Comment by WeiyiLi** [comment]
I tried the code in https://github.com/tensorpack/tensorpack/blob/master/examples/basics/export-model.py, but I didn't find where I should put "print(tensor)"...

**Comment by ppwwyyxx** [comment]
1. Find the input tensor in the code. In your case it's probably image: 
https://github.com/tensorpack/tensorpack/blob/274c754444ac35492e32b67764e2466b46f8a179/examples/ImageNetModels/shufflenet.py#L114
2. Add `print(image)`.
3. Find the output tensor in the code. In your case it may be logits:
https://github.com/tensorpack/tensorpack/blob/274c754444ac35492e32b67764e2466b46f8a179/examples/ImageNetModels/shufflenet.py#L153
4. Add `print(logits)`.
5. Use random strings as input names and output names and run the export code. It will print the names of image and logits.
